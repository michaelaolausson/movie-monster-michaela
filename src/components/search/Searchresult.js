import React from 'react';
import { Col, Row, Alert } from 'react-bootstrap'
import Movie from '../movie/Movie'


class Searchresult extends React.Component {

  render() {
    if (this.props.movies.results) {
      return(
        <Row>
        {this.props.movies.results.map((movie, index) => (
          <Movie id={movie.id} key={index} title={movie.title} year={movie.release_date} description={movie.overview} imagePath={movie.poster_path } />
      ))}
        </Row>
      )
    } else if (this.props.movies.results && this.props.movies.results.length === 0) {
       return(
        <Row>
          <Col lg={6}>
            <Alert variant="warning">
              Zero results. Zorry!
            </Alert>
          </Col>
        </Row>
      ) 
    } else {
      return (
        <Row></Row>
      )
    }
  }
}

export default Searchresult;
