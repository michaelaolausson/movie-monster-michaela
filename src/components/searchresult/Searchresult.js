import React from 'react';
import { ListGroup } from 'react-bootstrap'
import Movie from '../movie/Movie'


class Movielist extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {

    if (this.props.movies.results) {
      return(
        <ListGroup>
        {this.props.movies.results.map((movie, index) => (
          <Movie i={index} title={movie.title} year={movie.release_date} description={movie.overview} imagePath={movie.poster_path } />
      ))}
        </ListGroup>
      )

    } else {
      return (
        <ListGroup>
          <p className="text-muted">It's empty here!</p>
        </ListGroup>
      )
    }
  }
}

export default Movielist;
