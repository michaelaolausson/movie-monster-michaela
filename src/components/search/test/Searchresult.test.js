import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { shallow } from 'enzyme';
import Searchresult from '../Searchresult.js'
import MOCK_SEARCHRESULT from './Searchresult.testdata.js'

describe('Searchresult component', () => {
    it('renders correctly', () => {
        const wrapper = shallow(<Searchresult movies={MOCK_SEARCHRESULT} />);
        expect(wrapper).toMatchSnapshot();
    })
})