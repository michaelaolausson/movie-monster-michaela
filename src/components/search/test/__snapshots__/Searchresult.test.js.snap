// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Searchresult component renders correctly 1`] = `ShallowWrapper {}`;
